tests/ui_tests/app/page/admin/page.py

```diff
@@ -489,3 +510,83 @@ def click_delete_button(self):
+        AdminIntroLocators.intro_title,
+        AdminIntroLocators.intro_text,
+        AdminGroupsLocators.create_group_btn,
+        AdminGroupsLocators.delete_btn,
+        CommonTable.header,
+    ]
+
+    def click_create_policy_btn(self):
+        self.find_and_click(AdminPoliciesLocators.create_policy_btn)
+        self.wait_element_visible(AdminPoliciesLocators.AddPolicyPopup.block)
+
+    @allure.step('Create new policy')
+    def create_policy(
+        self, policy_name: str, description: Optional[str], role: str, users: Optional[str], groups: Optional[str]
+    ):
+        self.click_create_policy_btn()
+        self.send_text_to_element(AdminPoliciesLocators.AddPolicyPopup.FirstStep.name_input, policy_name)
+        if description:
+            self.send_text_to_element(AdminPoliciesLocators.AddPolicyPopup.FirstStep.description_input, description)
+        with allure.step(f"Select role {role} in popup"):
+            self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.FirstStep.role_select)
+            self.wait_element_visible(AdminPoliciesLocators.AddPolicyPopup.FirstStep.role_item)
+            for available_role in self.find_elements(AdminPoliciesLocators.AddPolicyPopup.FirstStep.role_item):
+                if available_role.text == role:
+                    available_role.click()
+                    break
+        assert users or groups, "There are should be users or groups in the policy"
+        if users:
+            with allure.step(f"Select users {users} in popup"):
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.FirstStep.users_select)
+                self.wait_element_visible(AdminPoliciesLocators.AddPolicyPopup.FirstStep.users_item)
+                available_users = self.find_elements(AdminPoliciesLocators.AddPolicyPopup.FirstStep.users_item)
+                for user in users.split(","):
+                    for available_user in available_users:
+                        if available_user.text == user:
+                            self.scroll_to(available_user)
+                            available_user.click()
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.FirstStep.users_select)
+        if groups:
+            with allure.step(f"Select groups {groups} in popup"):
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.FirstStep.group_select)
+                self.wait_element_visible(AdminPoliciesLocators.AddPolicyPopup.FirstStep.group_item)
+                available_groups = self.find_elements(AdminPoliciesLocators.AddPolicyPopup.FirstStep.group_item)
+                for group in groups.split(","):
+                    for available_group in available_groups:
+                        if available_group.text == group:
+                            self.scroll_to(available_group)
+                            available_group.click()
+                self.find_and_click(AdminPoliciesLocators.AddPolicyPopup.FirstStep.group_select)
```

Comment:
Looks quite the same at the first glance. Maybe we can add a separate function that takes locators for select and item, the text value of user/group, and do stuff?
And I'm not sure what will be if the user/group isn't presented in a list. If it fail silently, it may be difficult to understand and debug

Reply:
moved to a func